broker/lib/fabrik/DirectorManager.js

```diff
@@ -867,22 +867,66 @@ class DirectorManager extends BaseManager {
       return _.includes(['succeeded', 'failed', 'aborted'], state);
     }
 
+    let restoreMetadata;
     return this.agent
       .getRestoreLastOperation(agent_ip)
       .tap(lastOperation => {
         if (isFinished(lastOperation.state)) {
-          return this.agent
-            .getRestoreLogs(agent_ip)
-            .then(logs => this.backupStore
-              .patchRestoreFile(options, {
-                state: lastOperation.state,
-                logs: logs
-              })
-            );
+          return Promise.all([this.agent
+              .getRestoreLogs(agent_ip), this.backupStore
+              .getRestoreFile(options)
+            ])
+            .spread((logs, metadata) => {
+              restoreMetadata = metadata;
+              const restoreFinishiedAt = lastOperation.updated_at ? new Date(lastOperation.updated_at).toISOString() : new Date().toISOString();
+              const date_history = this.updateHistoryOfDates(
+                _.get(restoreMetadata, 'date_history'), restoreFinishiedAt);
+              return this.backupStore
+                .patchRestoreFile(options, {
+                  state: lastOperation.state,
+                  logs: logs,
+                  finished_at: restoreFinishiedAt,
+                  date_history: date_history
+                });
+            })
+            .tap(() => {
+              // Trigger schedule backup when restore is successful
+              if (lastOperation.state === CONST.OPERATION.SUCCEEDED) {
+                return this.scheduleBackup({
+                  instance_id: options.instance_guid,
+                  afterXminute: 3
+                });
+              }
+            });
         }
       });
   }
 
+  updateHistoryOfDates(arrayOfDates, isoDateToUpdate) {
+    let updatedHistory = arrayOfDates || [];
+    updatedHistory.push(isoDateToUpdate);
+    _.remove(updatedHistory, date => {
+      const twoMonthOlderDate = Date.now() - 1000 * 60 * 60 * 24 * 60;
+      return Date.parse(date) < twoMonthOlderDate;
+    });
+    return updatedHistory.sort();
+  }
+  scheduleBackup(opts) {
```

Comment:
There is scheduleBackup function in DirectorInstance as well. Not sure if we can modify and use the same, but else it might be confusing.

Reply:
renamed to scheduleBackup